src/main/java/org/springframework/retry/backoff/UniformRandomBackOffPolicy.java

```diff
@@ -134,7 +134,7 @@ public long getMaxBackOffPeriod() {
 	protected void doBackOff() throws BackOffInterruptedException {
 		try {
 			Long min = this.minBackOffPeriod.get();
-			long delta = this.maxBackOffPeriod.get() == this.minBackOffPeriod.get() ? 0
+			long delta = this.maxBackOffPeriod.get() <= this.minBackOffPeriod.get() ? 0
 					: this.random.nextInt((int) (this.maxBackOffPeriod.get() - min));
```

Comment:
I think we need to extract both `Supplier.get()` into local variables and use them afterwards.
I think the logical way is possibility to return different values for every `get()` call.
So, it is OK to have them different for each `backOff()` call, but not within a single one.
Otherwise it may lead to unexpected  behavior.
I mean we have that `Long min = this.minBackOffPeriod.get();` variable.
So, let's use that in the following code!
Plus let's extract `Long max = this.maxBackOffPeriod.get();`